pkg/storage/kubernetes/ipam.go

```diff
@@ -522,15 +531,40 @@ func IPManagementKubernetesUpdate(ctx context.Context, mode int, ipam *Kubernete
 						continue
 					}
 
-					ipforoverlappingrangeupdate = newip.IP
+					// reserve IP in overlappingIPRangeAllocation
+					err = overlappingrangestore.UpdateOverlappingRangeAllocation(requestCtx, mode, newip.IP, containerID, podRef)
+					// TODO: We should probably return a specific type of error in case k8s returns error AlreadyExsits, then exit early in case of any other error
+					if err != nil {
+						logging.Errorf("Failed to update overlappingRangeAllocation (%s). Continuing loop. %v", newip.IP, err)
+						continue
+					}
+
+					defer func() {
+						if err != nil {
```

Comment:
I dont think this will ever be hit? It only happens if err != nil but all previous conditions check for err != nil and continue or return.

Is there a reason for this defer func() at all? It does exactly the same as what was previously done above